pkg/providers/controller.go

```diff
@@ -262,6 +262,16 @@ func (c *Controller) initSharedInformers() *providertypes.ListerInformer {
 		ConfigMapLister:        configmapLister,
 	}
 
+	//wait for all informer has synecd
+	apisixFactory.Start(ctx.Done())
+	kubeFactory.Start(ctx.Done())
+	cache.WaitForCacheSync(ctx.Done(),
+		listerInformer.EpInformer.HasSynced,
+		listerInformer.SvcInformer.HasSynced,
+		listerInformer.SecretInformer.HasSynced,
+		listerInformer.PodInformer.HasSynced,
+		listerInformer.ApisixUpstreamInformer.HasSynced)
+
```

Comment:
It is wrong to start here, because the event handler function is not registered before starting, which will cause some events to be lost.

Reply:
`apisixFactory.Start(ctx.Done())` 
why is it needed？